src/Raven.Client/Documents/Session/DocumentSessionRevisions.cs

```diff
@@ -18,14 +19,43 @@ public DocumentSessionRevisions(InMemoryDocumentSessionOperations session) : bas
         {
         }
 
-        public List<T> GetFor<T>(string id, int start = 0, int pageSize = 25)
+        public List<T> GetFor<T>(string id, int start = 0, int pageSize = 25, bool metadataOnly = false)
         {
-            var operation = new GetRevisionOperation(Session, id, start, pageSize);
+            var operation = new GetRevisionOperation(Session, id, start, pageSize, metadataOnly);
 
             var command = operation.CreateRequest();
             RequestExecutor.Execute(command, Context, sessionInfo: SessionInfo);
             operation.SetResult(command.Result);
-            return operation.Complete<T>();
+            return operation.GetRevisionsFor<T>();
+        }
+
+        public List<MetadataAsDictionary> GetMetadataFor(string id, int start = 0, int pageSize = 25)
```

Comment:
What is the scenario that is enabled by allowing to get metadata of a range of revisions?

Reply:
To get their change vector and last modified date. This way you can load iterate and load just the revision that you want.